detox/android/detox/src/main/java/com/wix/detox/DetoxJUnitRunner.java

```diff
@@ -46,9 +46,9 @@ public void onDestroy() {
     }
 
     private void onBeforeAppOnCreate(Application app, Bundle arguments) {
-        if (DetoxInstrumentsManager.supports()) {
-            final String recordingPath = arguments.getString("detoxInstrumRecPath");
-            if (recordingPath != null) {
+        final String recordingPath = arguments.getString("detoxInstrumRecPath");
+        if (recordingPath != null) {
+            if (DetoxInstrumentsManager.supports()) {
```

Comment:
@valentynberehovyi can u double check that `getString` returns null if key doesn't exist?

Reply:
From the android framework description.
```
   /**
     * Returns the value associated with the given key, or null if
     * no mapping of the desired type exists for the given key or a null
     * value is explicitly associated with the key.
     *
     * @param key a String, or null
     * @return a String value, or null
     */
    @Nullable
    public String getString(@Nullable String key)
```

So null is expected here.